generator/internal/discovery_type_vertex.cc

```diff
@@ -44,6 +44,34 @@ absl::optional<std::string> CheckForScalarType(nlohmann::json const& j) {
   return absl::nullopt;
 }
 
+std::string FormatMessageDescription(nlohmann::json const& field,
+                                     int indent_level) {
+  std::string description;
+  if (field.contains("description")) {
+    description = absl::StrCat(
+        FormatCommentBlock(std::string(field["description"]), indent_level),
+        "\n");
+  }
+
+  if (field.contains("enum")) {
+    std::vector<std::pair<std::string, std::string>> enum_comments;
+    for (unsigned int i = 0; i < field["enum"].size(); ++i) {
+      if (field.contains("enumDescriptions") &&
+          i < field["enumDescriptions"].size()) {
+        enum_comments.emplace_back(std::string(field["enum"][i]),
+                                   std::string(field["enumDescriptions"][i]));
+      } else {
+        enum_comments.emplace_back(std::string(field["enum"][i]),
+                                   std::string());
+      }
+    }
+    absl::StrAppend(&description,
+                    FormatCommentKeyValueList(enum_comments, indent_level),
+                    "\n");
+  }
```

Comment:
I guess performance is not a concern, but continually re-evaluating `field["enum"]` at least smells a little.